File: backend/src/main/java/com/payroll/system/backend/service/SalaryService.java
package com.payroll.system.backend.service;

import com.payroll.system.backend.model.Salary;
import com.payroll.system.backend.model.SalaryStatus;
import com.payroll.system.backend.repository.SalaryRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class SalaryService {
    @Autowired
    private SalaryRepository salaryRepository;

    public List<Salary> getAllSalaries() {
        return salaryRepository.findAll();
    }

    public Optional<Salary> getSalaryById(Long id) {
        return salaryRepository.findById(id);
    }

    public List<Salary> getSalariesByEmployeeId(Long employeeId) {
        return salaryRepository.findByEmployeeId(employeeId);
    }

    public List<Salary> getSalariesByStatus(SalaryStatus status) {
        return salaryRepository.findByStatus(status);
    }

    public List<Salary> getSalariesByMonthAndYear(String month, int year) {
        return salaryRepository.findByMonthAndYear(month, year);
    }

    public Salary saveSalary(Salary salary) {
        // Calculate net pay before saving
        salary.calculateNetPay();
        return salaryRepository.save(salary);
    }

    public void deleteSalary(Long id) {
        salaryRepository.deleteById(id);
    }
}

File: src/services/salary-service.ts
import apiClient from './api-client';

interface Salary {
  id?: number;
  employeeId: number;
  amount: number;
  month: string;
  year: number;
  status: string;
  paymentDate?: string;
}

const SalaryService = {
  getAllSalaries: async () => {
    return await apiClient.get('/salaries');
  },

  getSalaryById: async (id: number) => {
    return await apiClient.get(`/salaries/${id}`);
  },

  getSalariesByEmployeeId: async (employeeId: number) => {
    return await apiClient.get(`/salaries/employee/${employeeId}`);
  },

  createSalary: async (salary: Salary) => {
    return await apiClient.post('/salaries', salary);
  },

  updateSalary: async (id: number, salary: Salary) => {
    return await apiClient.put(`/salaries/${id}`, salary);
  },

  deleteSalary: async (id: number) => {
    return await apiClient.delete(`/salaries/${id}`);
  }
};

export default SalaryService;
export type { Salary };